addons/pos_self_order/models/pos_config.py

```diff
@@ -341,3 +333,8 @@ def _modify_pos_restaurant_config(self):
     def _load_restaurant_data(self):
         super()._load_restaurant_data()
         self._modify_pos_restaurant_config()
+
+    @api.model
+    def get_kiosk_url(self, context):
+        record = self.browse(context['active_id']).read()[0]
+        return record['self_ordering_url']
```

Comment:
Kind of a poor usage of the ORM. Read without argument means to read all fields, but you only care about `self_ordering_url`. To create an api.model, but then browsing on the active_id to re-create a recordset it odd.

What are the reasons this cannot be:
```py
def get_kiosk_url(self):
    return self.self_ordering_url
```